galasa-ui/src/components/test-runs/OverviewTab.tsx

```
@@ -1,19 +1,46 @@
+"use client";
 /*
  * Copyright contributors to the Galasa project
  *
  * SPDX-License-Identifier: EPL-2.0
  */
 
-import React from "react";
+import React, { useEffect, useState } from "react";
 import styles from "@/styles/OverviewTab.module.css";
 import InlineText from "./InlineText";
 import { Tag } from "@carbon/react";
 import { RunMetadata } from "@/utils/interfaces";
 import { useTranslations } from "next-intl";
+import { Link } from "@carbon/react";
+import { Launch } from "@carbon/icons-react";
+import { getOneMonthAgo, getAWeekBeforeSubmittedTime } from "@/utils/timeOperations";
 
 const OverviewTab = ({ metadata }: { metadata: RunMetadata }) => {
   const tags = metadata?.tags || [];
   const translations = useTranslations("OverviewTab");
+  const [weekBefore, setWeekBefore] = useState("");
+  
+  const MONTH_AGO = getOneMonthAgo();
+
+  const OTHER_RECENT_RUNS = `/test-runs?testName=${metadata?.testName}&bundle=${metadata?.bundle}&group=${metadata?.group}&package=${metadata?.package}&from=${MONTH_AGO}`;
+  const RETRIES_FOR_THIS_TEST_RUN = `/test-runs?submissionId=${metadata?.submissionId}&from=${weekBefore}`;
+
+  useEffect(() => {
+
+    const validateTime = () => {
+
+      const validatedTime = getAWeekBeforeSubmittedTime(metadata?.rawSubmittedAt!);
+      if (validatedTime === "Invalid date") {
```

If the time is invalid, instead of setting an "Invalid date" value - maybe we either throw an error and catch it, or return a null value?